### ISSUE TITLE
Signal ground stations

### ISSUE THREAD

**Issue by KSAMissionCtrl** [ShotgunNinja/Kerbalism#97]
As an enhancement for the Signals system, being able to specify geographic points on Kerbin that can receive DSN comms would be nice. It could be that if the user chooses not to define any ground stations the whole of Kerbin is used as it is currently. Additionally rovers/aircraft in visual range of a ground station could transmit data there directly rather than through an orbiting satellite relay.

**Comment by eberkain** [comment]
The main use case for this is when doing stuff on kerbin and having to deal with LOS back to the ground station location.  Right now running remote operated craft on kerbin is too easy because the LOS does not matter, making kerbin comm satellites largely pointless, when they shouldn't be.  

I think there should be just a couple ground stations, maybe 3, beyond kerbin SOI you would get nearly 100% coverage, but closer to the surface, much less so.

**Comment by KSAMissionCtrl** [comment]
Yes, what you're saying is a good expansion on my reasoning. However there really shouldn't be any reason to limit the location or number of ground stations. It should be like RemoteTech, where you can just define your own although RT demands that KSC be a groundstation but since Kerbalism already functions fine without one it could still be an option for people who don't want to deal with this stuff. Really this is the only thing missing from the Signal system vs. RemoteTech. I personally prefer to handle signal delay on my own

**Comment by eberkain** [comment]
I would say that signal delay is probably the most unused feature of remote tech, its interesting, but just becomes annoying at some point.  Ground stations I think would add to the realism aspects of the mod instead of just having blanked 100% coverage at kerbin.   ShotgunNinja has been opposed to it in the past, maybe we can convince him to reconsider the feature.

**Comment by lordcirth** [comment]
Personally, I greatly prefer the simple "Kerbin is the antenna" solution.  Most trips on Kerbin surface I would consider to be early/beginner expeditions.  Let them be easy.  People can worry about full-coverage commsat constellations when they are landing on Duna.

**Comment by KSAMissionCtrl** [comment]
@lordcirth this isn't about removing the ability to have Kerbin be an antenna, so people who want to play like it already is have nothing to worry about. I already said in the absence of ground stations defined Kerbin would be a global receiver

**Comment by eberkain** [comment]
Yes, I also think it is a feature that should be able to be toggled within the signal feature.

**Comment by ShotgunNinja** [comment]
There are two cases: either
- (A) the vessel is on surface or inside atmosphere of home body, or
- (B) the vessel is elsewhere

In [A], assume connectivity is not an issue. It is not like in real life the DSN is used to communicate with airplanes, cars, or ships.

In [B], targeting individual DSN stations is intrinsically the same as targeting the planet itself. Assuming the DSN hardware coverage is uniform over the sphere.

So I don't really see the incentive in making the signal computation more involved than already is.

**Comment by KSAMissionCtrl** [comment]
Well, right now there is currently no reason to setup a communications satellite network around Kerbin. Everyone does it in the stock game, so I don't really see why they would complain about things being more involved here. It just seems like a silly thing to skimp on for a mod of this scope, which is putting way more of a demand on users for things that aren't even in the stock game

**Comment by ShotgunNinja** [comment]
I mean: it make the _computation_ of connectivity more involved. I'm all about making the _gameplay_ more involved for the player.

I think the arguments I made in previous post are valid ones. However if I can come up with a way to implement this without too much refactoring in the signal system, I'll do it. So let's keep this open for now.

**Comment by KSAMissionCtrl** [comment]
oh, oh - I completely read that wrong. Yes well, Kerbalism is indeed doing a lot of work behind the scenes so I can understand how performance might be an issue here.

**Comment by eberkain** [comment]
Could you treat the the current DSN system and a local connectivity system separately?  Give the DSN antenna a minimum range that is somewhere past LKO.  So anything on the surface or flying on kerbin doesn't get connectivity automatically, and instead has to rely on relay antennas.  Then add a ground station to KSC, which would essentially be a relay antenna that provides the master connection for control.
